### PR TITLE
add documentation for Maven + Bnd (migration, usage, ...)

### PR DESCRIPTION
**Pull request by maggu2810**
Adding the documentation to the repository allows to create and apply pull requests.

Migrated from: https://github.com/openhab/openhab2-addons/issues/5005#issue-415262659

### CODE REVIEW

File: docs/maven-bnd.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+* Offline mode builds
+* Less buildsystem file overhead
+* External dependencies per extension via the standard maven dependency system
+* Unit tests are now part of the same project (Eclipse allows to create a unit test for a class via GUI now)
+
+## How to migrate
+
+1. Move your addon to `/bundles/`. Adapt the `addons/pom.xml` and `/bundles/pom.xml`
+2. Copy over the `.classpath` from an already migrated bundle
+2. Copy over `.project` and adapt the `name`.
```
Comment (cweitkamp) [inline docs/maven-bnd.md]: Two times "2."

File: docs/maven-bnd.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+For integration tests you might need to run: `mvn clean install -DwithResolver`.
+
+Skip checks and tests if you just want to test the compilation process and perform the command from within the directory of the bundle that you want to build:
+
+`mvn clean install -DskipChecks -DskipTests -o` 
+
+## How to develop in the Eclipse IDE
+
+1. Install Bndtools in your Eclipse IDE. (Is automatically installed if you have used the Eclipse Installer for openHAB development.)
+1. Checkout the bnd based openHAB demo application: `git clone --depth=1 https://github.com/maggu2810/openhab-demo`.
```
Comment (cweitkamp) [inline docs/maven-bnd.md]: Two times "1."

File: docs/maven-bnd.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+   Replace "org.openhab.binding" with "org.openhab.addons.bundles" for your addon entry.
+   If you have external dependencies, you must list them here, too. An example:
+   ```xml
+      <feature name="openhab-binding-lgwebos" description="LG webOS Binding" version="${project.version}">
+           ...
+          <bundle dependency="true">mvn:org.apache.httpcomponents/httpclient-osgi/4.2.3</bundle>
+          <bundle start-level="80">mvn:org.openhab.binding/org.openhab.binding.lgwebos/${project.version}</bundle>
+      </feature>
+   ```
+   There are two types of external libraries. Those that are ready for OSGi (like the one above) and those that aren't. The latter ones are listet with a **wrap** prefix keyword like this:
```
Comment (cweitkamp) [inline docs/maven-bnd.md]:
```suggestion
   There are two types of external libraries. Those that are ready for OSGi (like the one above) and those that aren't. The latter ones are listed with a **wrap** prefix keyword like this:
```

File: docs/maven-bnd.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
+# Migration to a Maven + Bnd based build system
+
+We are in the process of changing the build system.
+We are not accepting new pull requests that are still using the old way from here on.
+
+## Background
+
+openHAB is a java project that is build around the OSGi framework. openHAB addons are OSGi bundles.
```
Comment (cweitkamp) [inline docs/maven-bnd.md]: I suggest to change "addon" to "add-on" throughout the whole file.

**Comment by cweitkamp** [comment]
And of course: Thank you!

File: docs/maven-bnd.md (hunk 1 above)
Comment (maggu2810) [reply docs/maven-bnd.md]: I agree that this should be fixed, but Github does not care about the number at all (IIRC). It uses an auto enumeration.

Regardless of Github's logic, I will change it.

File: docs/maven-bnd.md (hunk 2 above)
Comment (maggu2810) [reply docs/maven-bnd.md]: Same as above.

File: docs/maven-bnd.md (hunk 4 above)
Comment (maggu2810) [reply docs/maven-bnd.md]: done

File: docs/maven-bnd.md (hunk 3 above)
Comment (maggu2810) [reply docs/maven-bnd.md]: done

File: docs/maven-bnd.md (hunk 2 above)
Comment (maggu2810) [reply docs/maven-bnd.md]: done

File: docs/maven-bnd.md (hunk 1 above)
Comment (maggu2810) [reply docs/maven-bnd.md]: done

**Comment by maggu2810** [comment]
@cweitkamp all done

**Comment by maggu2810** [comment]
FYI: https://www.markdownguide.org/basic-syntax/#ordered-lists

**Comment by maggu2810** [comment]
@cweitkamp all done

File: docs/maven-bnd.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+   ```xml
+    <feature prerequisite="true">wrap</feature>
+    <bundle dependency="true">wrap:mvn:io.netty/netty-common/4.1.34.Final$Bundle-Name=Netty%20Common&amp;Bundle-SymbolicName=io.netty.netty-common&amp;Bundle-Version=4.1.34</bundle>
+   ```
+   Please ask for details and review if in doubt.
+
+There is a bash script to automatically migrate a binding. Run this script from the openhab2-addons directory: https://gist.github.com/Hilbrand/a7127d0fb7afd36124a78a248453ff03. The script doesn't migrate test projects and also locally added library in a lib folder need to migrated manually.
+
+#### Examples
+* for an add-on with lib directory: https://github.com/openhab/openhab2-addons/tree/master/bundles/org.openhab.binding.harmonyhub
```
Comment (wborn) [inline docs/maven-bnd.md]: It's no longer using this approach. :-)

Comment (maggu2810) [reply docs/maven-bnd.md]: Changed to allplay.

**Comment by maggu2810** [comment]
I am not really sure if this is the correct location to place that documentation.
The intention has been to be able to accept PRs and so contributions from others (e.g. small additions).

The part that explains the usage should perhaps moved to `openhab-docs` some time and I am not sure if the migration documentation should be added to the repo.

So, as long as its just a migration guide I will add it to a repo of mine until we find a better place.

WDYT?

**Pull request closed without merging**